tests/pytype_test.py

```diff
@@ -87,34 +109,60 @@ def communicate(self):
         return self.results
 
 
+def _get_relative(filename):
+    top = filename.find('stdlib/')
+    return filename[top:]
+
+
 def _get_module_name(filename):
-    """Converts a filename stdblib/m.n/module/foo to module.foo."""
-    return '.'.join(filename.split(os.path.sep)[2:]).replace(
+    """Converts a filename stdlib/m.n/module/foo to module.foo."""
+    return '.'.join(_get_relative(filename).split(os.path.sep)[2:]).replace(
         '.pyi', '').replace('.__init__', '')
 
 
-def pytype_test(args):
+def can_run(path, exe, *args):
+    exe = os.path.join(path, exe)
     try:
-        BinaryRun(['pytd', '-h']).communicate()
+        BinaryRun([exe] + list(args)).communicate()
+        return True
     except OSError:
+        return False
+
+def pytype_test(args):
+    dirs = get_project_dirs(args)
+    pytype_exe = os.path.join(dirs.pytype, 'pytype')
+    stdlib_path = os.path.join(dirs.typeshed, 'stdlib')
+
+    if not os.path.isdir(stdlib_path):
+        print('Cannot find typeshed stdlib at %s '
+            '(specify parent dir via --typeshed_location)' % stdlib_path)
+        return 0, 0
+
+    if can_run(dirs.pytype, 'pytd', '-h'):
+        pytd_exe = os.path.join(dirs.pytype, 'pytd')
+    elif can_run(dirs.pytype, 'pytd_tool', '-h'):
+        pytd_exe = os.path.join(dirs.pytype, 'pytd_tool')
+    else:
         print('Cannot run pytd. Did you install pytype?')
         return 0, 0
 
-    skip, parse_only = load_blacklist()
+    skip, parse_only = load_blacklist(dirs)
     wanted = re.compile(r'stdlib/.*\.pyi$')
     skipped = re.compile('(%s)$' % '|'.join(skip))
     parse_only = re.compile('(%s)$' % '|'.join(parse_only))
```

Comment:
It's not directly related to this PR, but while you're here: Could you handle the case of `skip` or `parse_only` being empty lists?

Reply:
done